---
title: "7-enrichment-analysis.RMD"
author: Ana BVA
date: "`r BiocStyle::doc_date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

## Introduction

<div class="alert alert-info">
  <strong>Aim:</strong> To merge data from individual DE analysis
</div>

We previously had run the following scripts:

- 0: Data selection `/0-Data-selection.RMD`

- 1: Dowload RAW data `/1-download_raw-data.RMD`

- 2: Normalizing data `/2-normalizing-data.RMD`

- 3: Curating data `/3-curatig-data.RMD`

- 4: Differential gene expression analysis  `/4-DE.RMD`

- 5: Merging data `/5-merge-data.RMD`

- 6: Merging data `/6-meta-analysis.RMD`




## Input

This script needs the following files:

<div class="alert alert-warning">
  <strong>Data:</strong> Meta-analysis results in `csv` files  
</div>

## Analysis

### Setup

For running the script, type:

```{bash, eval = F}
ssh -X aaltamirano@dna.lavis.unam.mx
qrsh
cd /mnt/Genoma/amedina/DataDNA/R-projects/Meta-analysis_COPD/output_data/.out
 module load r/4.0.1 

## OR
ssh ana@10.200.0.42
cd /home/ana/DataDNA/R-projects/Meta-analysis_COPD/output_data/.out

nohup R -e "rmarkdown::render(here::here('vignettes/8-enrichment-pathways.RMD'))" > 8-enrichment-pathways.RMD.out &
```

The script can be found in: `r getwd() `. And the directories were set using
`R/setup.R` (e.i. DATA, OUPUT, DOWNLOAD) as a small functions that will paste the names into complete paths.


```{r setup}
knitr::opts_knit$set(root.dir = here::here())
source(here::here("Tissue/R/setup.R"))
```

And this analysis is run in: `r getwd() `


### Libraries

```{r libraries, message=FALSE, warning=FALSE}
library(hypeR)
library(WGCNA)
library(tidyverse)
library(reactable)
library(ggpubr)
```


### Import data

We will import DE results from all experiments. 

```{r}
meta <- read_csv(OUTPUT("2021-02-02_Step6_meta-analysis.csv"))
meta
```



### Ranking and weighted values

```{r , class.source="bg-info",class.output="bg-info" }
qval_cutoff <- 0.05

up_df <- meta %>%
  filter(qval.fixed <= qval_cutoff) %>%
  filter(is.na(tau2) == F) %>%
  #filter(TE.fixed > 0) %>%
  dplyr::arrange(desc(TE.fixed))
  #dplyr::arrange(estimate)

up_signature <- as.vector(up_df$TE.fixed)
names(up_signature) <- up_df$X1
head(up_signature)


down_df <- meta %>%
  filter(qval.fixed <= qval_cutoff) %>%
  filter(is.na(tau2) == F) %>%
  #filter(TE.fixed < 0) %>%
  dplyr::arrange((TE.fixed))
  #dplyr::arrange(estimate)

down_signature <- as.vector(down_df$TE.fixed)
names(down_signature) <- down_df$X1
head(down_signature)
```

### Enrichment analysis

Then we will check KEGG, REACTOME and some databases from EnrichR


```{r, eval=FALSE}
# KEGG
KEGG <- hyperdb_rgsets("KEGG", "92.0")

#HALLMARL
HALLMARK <- msigdb_gsets(species="Homo sapiens", category="H",clean = T)

#ENRICHR
TF_TJ <- enrichr_gsets("TRANSFAC_and_JASPAR_PWMs")

enrichR<- enrichr_gsets("Jensen_DISEASES")

enrichR<- enrichr_gsets("Jensen_TISSUES")

enrichR<- enrichr_gsets("Chromosome_Location")

enrichR<- enrichr_gsets("WikiPathways_2019_Human")

enrichR<- enrichr_gsets("ClinVar_2019")


```



```{r}
# KEGG
KEGG <- hyperdb_rgsets("KEGG", "92.0")

#HALLMARL
HALLMARK <- msigdb_gsets(species="Homo sapiens", category="H",clean = T)

DB <- KEGG
DB <- HALLMARK
  # Ranked Signature using the estimator from the meta analysis (FC)
  hyp_obj <- hypeR(names(up_signature),DB, test="hypergeometric", background=nrow(meta), fdr=0.5, plotting=TRUE)
  hyp_obj$plots[[1]]
  hyp_obj
  hyp_dots(hyp_obj) + theme_minimal()
  
  # Weighted Signature using as ranking the estimator and the weighted the qvalue
  hyp_obj <- hypeR(up_signature, DB, test="kstest", background=nrow(meta),fdr=0.5, plotting=TRUE)
  hyp_obj$plots[[1]]
  hyp_dots(hyp_obj) + theme_minimal(14)
  
  # Show interactive table
  #hyp_show(hyp_obj)
  
  # Plot enrichment map
  #hyp_emap(hyp_obj)
  
  # Plot hiearchy map
  #hyp_hmap(hyp_obj)
  
  # Generate markdown report
  #hyp_to_rmd(hyp_obj,"hypr-KEGG-genesets.rmd")
  #hyp_to_rmd(hyp_obj,"hypr-TRANSFAC_and_JASPAR_PWMs.rmd")
```


```{r}
  # Ranked Signature using the estimator from the meta analysis (FC)
  hyp_obj <- hypeR(names(down_signature),DB, test="hypergeometric", background=nrow(meta), fdr=0.5, plotting=TRUE)
  hyp_obj$plots[[1]]
  hyp_obj
  
```

```{r}

# UP KEGG
up_kegg_obj <- hypeR(up_signature, 
                   KEGG, 
                   test="kstest", 
                   background=nrow(meta),
                   fdr=0.1, plotting=TRUE)

 up_kegg <-  hyp_dots(up_kegg_obj) + theme_minimal() + xlab("KEGG")
 
 # UP HALLMARL
 up_hall_obj <- hypeR(up_signature, 
                   HALLMARK, 
                   test="kstest", 
                   background=nrow(meta),
                   fdr=0.1, plotting=TRUE)

 up_hall <-  hyp_dots(up_hall_obj) + theme_minimal() + xlab("HALLMARL")
 
 # UP KEGG
down_kegg_obj <- hypeR(down_signature, 
                   KEGG, 
                   test="kstest", 
                   background=nrow(meta),
                   fdr=0.1, plotting=TRUE)

 down_kegg <-  hyp_dots(down_kegg_obj) + theme_minimal() + xlab("KEGG")
 
 
  # DOWN HALLMARL
 down_hall_obj <- hypeR(down_signature, 
                   HALLMARK, 
                   test="kstest", 
                   background=nrow(meta),
                   fdr=0.1, plotting=TRUE)

 down_hall <-  hyp_dots(down_hall_obj) + theme_minimal() + xlab("HALLMARL")
 
 
ggpubr::ggarrange(up_kegg,down_kegg,up_hall,down_hall, 
                   common.legend = T,
                   labels = c("UP","DOWN","UP","DOWN"),
                   ncol = 2,nrow = 2) 

  
 
```




## Output

This script produces the following data, and can be found in `r getwd() ` 

<div class="alert alert-success">
  <strong>Figures in the vignette:</strong> Enrichment analysis with plots
</div>

## Session Info

```{r sessionInfo}
sessionInfo()
```





