---
title: "7-enrichment-analysis.RMD"
author: Ana BVA
date: "`r BiocStyle::doc_date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

## Introduction

<div class="alert alert-info">
  <strong>Aim:</strong> To merge data from individual DE analysis
</div>

We previously had run the following scripts:

- 0: Data selection `/0-Data-selection.RMD`

- 1: Dowload RAW data `/1-download_raw-data.RMD`

- 2: Normalizing data `/2-normalizing-data.RMD`

- 3: Curating data `/3-curatig-data.RMD`

- 4: Differential gene expression analysis  `/4-DE.RMD`

- 5: Merging data `/5-merge-data.RMD`

- 6: Merging data `/6-meta-analysis.RMD`




## Input

This script needs the following files:

<div class="alert alert-warning">
  <strong>Data:</strong> Meta-analysis results in `csv` files  
</div>

## Analysis

### Setup

For running the script, type:

```{bash, eval = F}
ssh -X aaltamirano@dna.lavis.unam.mx
qrsh
cd /mnt/Genoma/amedina/DataDNA/R-projects/Meta-analysis_COPD/output_data/.out
 module load r/4.0.1 

## OR
ssh ana@10.200.0.42
cd /home/ana/DataDNA/R-projects/Meta-analysis_COPD/output_data/.out

nohup R -e "rmarkdown::render(here::here('vignettes/8-enrichment-pathways.RMD'))" > 8-enrichment-pathways.RMD.out &
```

The script can be found in: `r getwd() `. And the directories were set using
`R/setup.R` (e.i. DATA, OUPUT, DOWNLOAD) as a small functions that will paste the names into complete paths.


```{r setup}
knitr::opts_knit$set(root.dir = here::here())
source(here::here("Tissue/R/setup.R"))
```

And this analysis is run in: `r getwd() `


### Libraries

```{r libraries, message=FALSE, warning=FALSE}
library(hypeR)
library(tidyverse)
library(reactable)
library(ggpubr)
library(fgsea)
library(data.table)
library(grid)
```


### Import datahyp

We will import DE results from all experiments. 

```{r}
meta <- read_csv(OUTPUT("2021-03-01_Step6_meta-analysis.csv"))
meta

```



### Pathways

Then we will check KEGG, REACTOME and some databases from EnrichR

Other pathways
```{r}
KEGG <- hyperdb_rgsets("KEGG", "92.0")
HALLMARK <- msigdb_gsets(species="Homo sapiens", category="H",clean = T)
TF_TJ <- enrichr_gsets("TRANSFAC_and_JASPAR_PWMs")
JD<- enrichr_gsets("Jensen_DISEASES")
JT<- enrichr_gsets("Jensen_TISSUES")
goB<- enrichr_gsets("GO_Biological_Process_2018")
goM<- enrichr_gsets("GO_Molecular_Function_2018")
goC<- enrichr_gsets("GO_Cellular_Component_2018")

#enrichR<- enrichr_gsets("GWAS_Catalog_2019")
#enrichR<- enrichr_gsets("Reactome_2016")
#enrichR<- enrichr_gsets("WikiPathways_2019_Human")
```


#GSEA

Function to plot fGSEA figures

```{r}
gsea.plot <- function(examplePathways = examplePathways, 
                      signature = signature,
                      i = 1,
                      cpadj = 0.1){
# 
fgseaRes <- fgsea(pathways = examplePathways, 
                  stats    = signature,
                  eps      = 0.0,
                  minSize  = 15,
                  maxSize  = 500)

#print(head(fgseaRes[order(pval), ]))

## Table order by padj
fgo <- as_tibble(fgseaRes) %>%
  filter(padj <= cpadj) %>%
  arrange(-padj) %>% 
  mutate(name = factor(pathway, levels=pathway))

# plot with top pathways
gg <- ggplot(fgo) +
  aes(x = name, y = padj) +
  geom_point( size=5, color="#5abdc4") +
  labs(x = "Pathways") +
  coord_flip() +
  geom_hline(yintercept=0.01, linetype="dashed", color = "red")+
  theme_minimal(10)

# one plot
x <- fgo %>% 
  arrange(padj)

x <- x$pathway[i]

one <- plotEnrichment(examplePathways[[x]],
               signature) + labs(title=x)
#dev.off()

# top pathways
topPathwaysUp <- fgseaRes[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- fgseaRes[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
top <- plotGseaTable(examplePathways[topPathways], signature, fgseaRes, 
              gseaParam=0.5, render = F)
#dev.off()

# collapsed pathways
collapsedPathways <- collapsePathways(fgseaRes[order(pval)][padj < 0.01], 
                                      examplePathways, signature)
mainPathways <- fgseaRes[pathway %in% collapsedPathways$mainPathways][
                         order(-NES), pathway]
com <- plotGseaTable(examplePathways[mainPathways], signature, fgseaRes, 
              gseaParam = 0.5, render = F)
#dev.off()

return(list(table = fgo,gg = gg, one = one, top = top, collapse = com))

}

```

Setup the data

```{r}
### Selecting genes
# Rank based on https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2019/RNAseq/html/06_Gene_set_testing.html

qval_cutoff <- 0.05

up_df <- meta %>%
  filter(num_exp > 3) %>% 
  mutate(rank = -log10(pval.random) * sign(TE.random)) %>% 
  dplyr::arrange(desc(rank)) 

signature <- as.vector(up_df$rank)
names(signature) <- up_df$X1
head(signature)
```

We do the enrichment analysis and plots

```{r}
k <- gsea.plot(examplePathways = KEGG$genesets, signature = signature, cpadj = 0.05, i = 2)
h <- gsea.plot(examplePathways = HALLMARK$genesets, signature = signature, cpadj = 0.05)
jt <- gsea.plot(examplePathways = JT$genesets, signature = signature, cpadj = 0.05)
gob <- gsea.plot(examplePathways = goB$genesets, signature = signature, cpadj = 0.01)
goc <- gsea.plot(examplePathways = goC$genesets, signature = signature, cpadj = 0.05)
gom <- gsea.plot(examplePathways = goM$genesets, signature = signature, cpadj = 0.05)

ggarrange(k$gg,h$gg,#jt$gg,gob$gg,goc$gg,gom$gg,
          labels = c("KEGG", "HALLMARK"), #"Jensen TISSUES",
                     #"GO: Biology", "GO: Celular", "GO: Molecular"),
          vjust = 0.5)
#
#ggarrange(k$one,h$one,jt$one,gob$one,goc$one,gom$one)
#dev.off()
#
#gridExtra::grid.arrange(k$top,h$top,jt$top,
#          gob$top,goc$top, gom$top,
#          ncol = 3)
#dev.off()
#
#gridExtra::grid.arrange(k$collapse,h$collapse,jt$collapse,
#          gob$collapse,goc$collapse, gom$collapse,
#          ncol = 3)
#dev.off()

grid.draw(h$top)
dev.off()

gridExtra::grid.arrange(k$one, 
                        k$top, 
                        ncol = 1,
                        heights = c(2,5),
                        nrow = 2)

gridExtra::grid.arrange(#gob$gg, #goc$gg,  gom$gg,
                        gob$one, #goc$one,gom$one,
                        gob$top, #goc$top,gom$top,
                        vp=viewport(width=.6),
                        #labels = c("GO: Biology","GO: Celular", "GO: Molecular"),
                        ncol = 1,
                        heights = c(2,5),
                        nrow = 2)

 
```


## Output

This script produces the following data, and can be found in `r getwd() ` 

<div class="alert alert-success">
  <strong>Figures in the vignette:</strong> Enrichment analysis with plots
</div>

## Session Info

```{r sessionInfo}
sessionInfo()
```





