---
title: "Differential gene expression analysis of COPD data"
author: Ana BVA
date: "`r BiocStyle::doc_date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

## Introduction

<div class="alert alert-info">
  <strong>Aim:</strong> To pre-process raw data from *GEO* using RMA and to 
  perform a differential gene expression analysis comparing COPD vs CONTROL group
</div>


### COPD

Chronic Obstructive Pulmonary Disease (COPD) is characterized by emphysema and 
chronic bronchitis, it's diagnosed using spirometry and clinical information
which lead to a heterogeneous COPD patients. The ranking of non commutable 
diseases from the WHO estimates that COPD is in the top of mortality causes and 
tobacco is the main risk factor but different genetic variants have been
associated with this disease. 

Different researchers have analyzed COPD transcriptomics using high-throughput data
such as microarrays and RNA-seq. We belive it would be relevant to unravel a 
robust gene expression signature for COPD patients regardless if it is from different
experiments or laboratories. 

### Background 

This script is part of a meta-analysis and it amis to
determine a common deferentially expressed genes. The input is an `ExpressionSet`
object with data already curated to have a phenotype column named 'DISEASE' 
and the levels are `CONTROL` or `COPD`. 

The results will be `logFC`, `adjusted p-values`, etc. per experiment and they 
will be used to perfom a meta-analysis.

We want transcriptomic experiments from [GEO](https://www.ncbi.nlm.nih.gov/geo/) that have:

- Lung tissue samples

- COPD vs CONTROL group


## Input

This script needs the following files:

<div class="alert alert-success">
  <strong>Data 1:</strong> GSE experiments   
</div>


### Setup

```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```


The script can be found in: `r getwd() `


```{r setup, class.source="bg-info"}
PATH = here::here()
DATA_DIR = file.path(PATH,"data")
OUTPUT_DIR = file.path(PATH,"output_data")
DOWNLOAD = file.path(PATH,"download")
FIG_DIR = file.path(PATH,"fig")
TODAY = Sys.Date()

knitr::knit_hooks$set(timeit = local({
  now = NULL
  function(before, options) {
    if (before) {
      now <<- Sys.time()
    } else {
      runtime = difftime(Sys.time(), now)
      now <<- NULL
      # use options$label if you want the chunk label as well
      paste('Time for this code chunk:', as.character(runtime))
    }
  }})
)

knitr::opts_knit$set(root.dir = PATH)
knitr::opts_chunk$set(echo = TRUE,
                      timeit=TRUE,
                      warning=FALSE,
                      attr.output='style="max-height: 500px;"')
```

And the analysis is run in: `r getwd() `

### Libraries

```{r libraries-install,message=FALSE, warning=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
      install.packages("BiocManager")
  }

packages <- c("knitr",
          "oligo",
          "tidyverse",
          "limma",
          "SummarizedExperiment",
          "DESeq2",
          "stringr",
          "org.Hs.eg.db",
          "AnnotationDbi",
          "vsn",
          "recount")

# for(l in packages){
#  if (!requireNamespace(l, quietly = TRUE)) {
#    BiocManager::install(l)}
#}
```


```{r libraries, message=FALSE, warning=FALSE}
lapply(packages, library, character.only = TRUE)
```

### Experiments

We selected experiments from lung samples of COPD patients 
and that also have a control group to compare. These experiments are described 
in the following table:


```{r experiments}
gse_table <- read.csv(file.path(OUTPUT_DIR,"2020-09-03-GSE_Summary.csv"), row.names = 1)
kable(gse_table, caption = "GSE information")
```

The experiment **GSE57148** is a RNA-seq experiment, and we do not need to normalize
the data using RMA but we will download counts from [ReCount2](https://jhubiostatistics.shinyapps.io/recount/).

We already downloaded the data using `GEOquery` and parsed the information using
`curateing_COPDinfo.R`script. So we can read the `.RDS` object into our R session

```{r}
geo <- readRDS(file.path(DATA_DIR,"2020-09-GSE_LungTissue-CURATED.RDS"))
names(geo)
```



## Local functions

### **DE**: Differential expression analysis

Using this funtion, we get a table with differential expression gene results using
`limma` package for fitting a linear model to get genes differentially expressed
between a "Control" and a "COPD" group. 

<div class="alert alert-warning">
  <strong>Input:</strong> GSE ID, optional: *colCOPD* is the column name in which 
  the information of disease status can be found, *coeff* will show results of 
  contrast with coeffitient found in possiton 2   
  <strong>Output:</strong> Table of differential expression results with all
  genes 
</div>


```{r DE}
DE <- function(ExpressionSet,colCOPD="DISEASE",coeff= "DISEASECOPD"){
     # it creates the design matrix and performs limma
     fit <- lmFit(ExpressionSet, model.matrix(as.formula(paste("~ 1 +", colCOPD)),
                                              data = pData(ExpressionSet)))
     # eBayes in lmFit model
     ebf <- eBayes(fit)
     print(colnames(coef(fit)))
     # It gets the genes with the p-values
     volcanoplot(ebf,coef = coeff,highlight=20, pch=20)
     res <- topTable(ebf, number = Inf, p.value = 1, coef = coeff,confint=T)
     # It formats in a tibble
     res <- as_tibble(res,rownames="rownames")
}

```


## `r names(geo)[1] `

Names will be different but it is **important** to check that "Control" group
is the first level. If need it re-level groups.

### Differential expression analysis

Using `DE()` function (described above), we performed a lineal regression model 
to calculate the logarithm fold change of all genes between a "Control" and a
"COPD" group. We also rename colnames adding the GSE ID at the end and finally, 
we save the output in a `.CSV` file. 

```{r}
i <- 1
g <- geo[[i]][[2]]
boxplot(g)
hist(g)

de <- DE(g,colCOPD = "DISEASE + PATIENT")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de1 <- de
```


## `r names(geo)[2] `

```{r}
i <- 2
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de2 <- de
```


## `r names(geo)[3] `

```{r}
i <- 3
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de3 <- de
```



## `r names(geo)[4] `

This experiment is RNAseq, the data will be download from Recount2. Iâ€™m following Recount2 vignette
Using `DESeq2`package we calculated DEG.  

```{r}
## Specify design and switch to DESeq2 format
i <- 4
rse <- geo[[i]][[1]]

 dds <- DESeqDataSet(rse, ~ DISEASE)
 
 ## Perform DE analysis
 dds <- DESeq(dds)
 boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2)
 
 ntd <- normTransform(dds)
 meanSdPlot(assay(ntd))
 
 res <- results(dds)
 plotMA(res, ylim=c(-2,2))
 
 # Calculates de CI
 res$error <- qnorm(0.975)*res$lfcSE
 res$CI.L <- res$log2FoldChange-res$error
 res$CI.R <- res$log2FoldChange+res$error
 
 res
 
 ## Extract Gencode gene ids
 gencode <- gsub('\\..*', '', names(recount_genes))
 
## Find the gene information we are interested in
gene_info <- AnnotationDbi::select(org.Hs.eg.db, gencode, c('SYMBOL', 'ENSEMBL'), 'ENSEMBL')

r <- as_tibble(res, rownames="rownames")
r$rownames <- gsub("\\..*","",r$rownames)
r <- full_join(r,gene_info, by=c("rownames"="ENSEMBL")) 
r$GENE.SYMBOL <- r$SYMBOL
r$logFC <- r$log2FoldChange

colnames(r) <- str_c(colnames(r),"_",names(geo[i]))
colnames(r)


write_csv(r,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv"))

de4 <- r
```



## `r names(geo)[5] `

```{r}
i <- 5
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de5 <- de
```


## `r names(geo)[6] `

```{r}
i <- 6
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de6 <- de
```

## `r names(geo)[7] `

```{r}
i <- 7
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de7 <- de
```

## `r names(geo)[8] `

```{r}
i <- 8
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de8 <- de
```

## `r names(geo)[9] `

```{r}
i <- 9
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de9 <- de
```



## `r names(geo)[10] `

```{r}
i <- 10
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de10 <- de
```



## `r names(geo)[11] `

```{r}
i <- 11
g <- geo[[i]][[1]]
boxplot(g)
hist(g)
de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de11 <- de
```




## `r names(geo)[11] `

```{r eval=FALSE, include=FALSE}
i <- 12
g <- geo[[i]][[1]]
boxplot(g)
hist(g)

de <- DE(g,colCOPD = "DISEASE")

colnames(de) <- str_c(colnames(de),"_",names(geo)[i])
colnames(de)

write_csv(de,
          path=str_c(OUTPUT_DIR,"/DE/",TODAY,"_TableGenes_",names(geo[i]),".csv")
          )
de12 <- de
```




############################




## Output

This script produces the following data, and can be found in `r getwd() ` 

<div class="alert alert-success">
  <strong>Tables with DE results:</strong> Tables with log fold change and p-values 
  calculated   
  <strong>Table of merged results:</strong> Table with all DE results 
</div>

## Session Info

```{r sessionInfo, attr.output='style="max-height: 500px;"'}
sessionInfo()
```





